internal/extrafiles/extra_files_test.go

```diff
@@ -4,15 +4,45 @@ import (
 	"testing"
 
 	"github.com/goreleaser/goreleaser/pkg/config"
+	"github.com/goreleaser/goreleaser/pkg/context"
 	"github.com/stretchr/testify/require"
 )
 
+func TestTemplate(t *testing.T) {
+	globs := []config.ExtraFile{
+		{Glob: "./testdata/file{{ .Env.ONE }}.golden"},
+	}
+
+	ctx := context.New(config.Project{})
+	ctx.Env["ONE"] = "1"
+
+	files, err := Find(ctx, globs)
+	require.NoError(t, err)
+	require.Len(t, files, 1)
+
+	require.Equal(t, "testdata/file1.golden", files["file1.golden"])
+}
+
+func TestBadTemplate(t *testing.T) {
+	globs := []config.ExtraFile{
+		{Glob: "./testdata/file{{ .Env.NOPE }}.golden"},
+	}
+
+	ctx := context.New(config.Project{})
+
+	files, err := Find(ctx, globs)
+	require.Empty(t, files)
+	require.EqualError(t, err, `failed to apply template to glob "./testdata/file{{ .Env.NOPE }}.golden": template: tmpl:1:22: executing "tmpl" at <.Env.NOPE>: map has no entry for key "NOPE"`)
+}
+
 func TestShouldGetSpecificFile(t *testing.T) {
 	globs := []config.ExtraFile{
+		{},                        // empty glob, will be ignored
+		{Glob: "./testdata/sub3"}, // will get a file1.golden as well, but will be overriden
```

Comment:
To appease the linter:

```suggestion
		{Glob: "./testdata/sub3"}, // will get a file1.golden as well, but will be overridden
```